Training/Training/SortAndSearch.c

```diff
@@ -3,24 +3,38 @@
 // Copyright (c) Gokul TS @Metamation India.
 // ------------------------------------------------------------------
 // SortAndSearch.c
-// C Program for Bubble sort and binary search algorithm
+// C Program for Bubble sort,Insertion sort and binary search algorithm
 // ------------------------------------------------------------------------------------------------
 #include "SortAndSearch.h"
 
 void BubbleSort (int arr[], int size) {
-   for (int i = 0; i < size - 1; i++)
+   int swapped;
+   for (int i = 0; i < size - 1; i++) {
+      swapped = 0;
       for (int j = 0; j < size - i - 1; j++)
          if (arr[j] > arr[j + 1]) {
             arr[j] += arr[j + 1];

```

Comment:
In this case, there might be an integer overflow error. So, using a temp variable for this will be an efficient and reliable method.